### PR TITLE
finished all features

### PR DESCRIPTION
**Pull request by Fedirrr**

### CODE REVIEW

File: src/api/movies.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import {sliderArray, moviesArray} from "../components/mock";
+
+
+export const fetchSliderDetailsById =  async (sliderId: string) => {
```
Comment (VladDrawer) [inline src/api/movies.ts:4]: please remove async

File: src/api/movies.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    return sliderArray.find(({id}) => id === sliderId)
+};
+
+export const fetchMovies = async () => {
```
Comment (VladDrawer) [inline src/api/movies.ts:8]: please remove async

File: src/components/AboutCompany.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import {selectCompany, setSCompanyDetails,} from "../store/slices/aboutCompanySlice";
+import {IViewers, viewersArray} from "./mock";
+import cssStyles from "../styles/styles";
+
+const AboutCompany = () => {
+    const {id} = useParams();
+    const comp = useSelector(selectCompany)
+    const [company, setCompany] = useState<IViewers | null>(null)
+    const dispatch = useDispatch()
+    const {aboutCompany} = cssStyles
```
Comment (VladDrawer) [inline src/components/AboutCompany.tsx:13]: add enters in all file

File: src/components/AboutCompany.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    useEffect(() => {
+        if (id && comp) {
+            const foundCompany = comp.find((movie) => movie.id === id) || null
+            setCompany(foundCompany)
+        }
+    }, [id, comp]);
+    return (
+        <div>
+
```
Comment (VladDrawer) [inline src/components/AboutCompany.tsx:27]: bad code style

File: src/components/AboutCompany.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+            {
+                company ? (
+                        <aboutCompany.CompanyContainer>
+                            <aboutCompany.BackgroundImageContainer>
+                                <aboutCompany.BackgroundImage
+                                    src={company.bgCompanyImg}
+                                    alt=""/>
```
Comment (VladDrawer) [inline src/components/AboutCompany.tsx:34]: bad code style

File: src/components/AboutCompany.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+                                <aboutCompany.BackgroundImageTextWrap
+                                >
+                                    <h1>{company.companyTitle}</h1>
+                                </aboutCompany.BackgroundImageTextWrap>
+                            </aboutCompany.BackgroundImageContainer>
+                            <img src="" alt=""/>
```
Comment (VladDrawer) [inline src/components/AboutCompany.tsx:40]: why did you add an empty image?

File: src/components/Details.tsx [hunk 7]
```diff
@@ -1,50 +1,51 @@
...
 
     const [isFavoriteAdded, setIsFavoriteAdded] = useState(false);
     const [movie, setMovie] = useState<IMovie | null>(null);
-
-
-    useEffect(() => {
-        const foundListItem = watchItems.find(item => item.id === id);
-        setIsFavoriteAdded(!!foundListItem)
-    }, [id, watchItems]);
+    const [modal, setModal] = useState(false);
```
Comment (VladDrawer) [inline src/components/Details.tsx:18]: all boolean variables should have names like this: isModal, isSomething, etc...

File: src/components/Header.tsx [hunk 8]
```diff
@@ -48,7 +48,7 @@ const Header = () => {
                                onKeyPress={handleKeyPress}
                 />
                 <header.FormBtn onClick={handleSearch}
-                disabled={searchTerm ? false : true}>
+                                disabled={searchTerm ? false : true}>
```
Comment (VladDrawer) [inline src/components/Header.tsx:51]: disabled={!searchTerm}>

File: src/components/Home.tsx [hunk 9]
```diff
@@ -3,23 +3,41 @@ import {useDispatch} from "react-redux";
...
 import Viewers from "./Viewers";
 import HomeMovies from "./HomeMovies";
-import {setMovies} from "../store/slices/slice";
-import {moviesArray} from "./mock";
+import {setMovies} from "../store/slices/movieSlice";
 import cssStyles from "../styles/styles";
+import {fetchMovies, fetchSliderDetailsById} from "../api/movies";
+import {useParams} from "react-router-dom";
+import {setSliderDetails} from "../store/slices/sliderDetailsSlice";
+import {setSCompanyDetails} from "../store/slices/aboutCompanySlice";
```
Comment (VladDrawer) [inline src/components/Home.tsx:11]: Bad name

File: src/components/Modal.tsx [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+                    </div>
+
+                    <modal.TitleBtn
+                        onClick={() => {
+                            closeModal(false);
+                        }}>
+                        X
+                    </modal.TitleBtn>
+                </modal.Title>
+                <iframe width="100%" height="100%" src={moviesList.trailerSource}
```
Comment (VladDrawer) [inline src/components/Modal.tsx:29]: a lot of inline styles are used here. Please create a separate file for styles

File: src/store/slices/watchListSlice.ts [hunk 11]
```diff
@@ -0,0 +1,34 @@
+import { createSlice } from "@reduxjs/toolkit";
+import { IMovie } from "../../components/mock";
+import { IState } from "../types/redux";
+
+const getLocalStorageData = () => {
+    const localStorageData = localStorage.getItem("watchlist");
+    return localStorageData ? JSON.parse(localStorageData) : [];
```
Comment (VladDrawer) [inline src/store/slices/watchListSlice.ts:7]: every json parse should be wrapped in try catch block

File: src/api/movies.ts (hunk 1 above)
Comment (Fedirrr) [reply src/api/movies.ts:4]: +

File: src/api/movies.ts (hunk 2 above)
Comment (Fedirrr) [reply src/api/movies.ts:8]: +

File: src/components/AboutCompany.tsx (hunk 5 above)
Comment (Fedirrr) [reply src/components/AboutCompany.tsx:34]: +

File: src/components/AboutCompany.tsx [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+
+            {
+                company ? (
+                        <aboutCompany.CompanyContainer>
+                            <aboutCompany.BackgroundImageContainer>
+                                <aboutCompany.BackgroundImage
+                                    src={company.bgCompanyImg}
+                                    alt=""/>
+                                <aboutCompany.BackgroundImageTextWrap
+                                >
```
[the remark this answers is not in the document]
Comment (Fedirrr) [reply src/components/AboutCompany.tsx:36]: +

File: src/components/AboutCompany.tsx [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+                            <img src="" alt=""/>
+                            <div className="block2_text_block" style={{
+                                padding: "50px"
+                            }}>
+                                {company.companyDescription}
+                            </div>
+                        </aboutCompany.CompanyContainer>
+                    ) :
+                    <div
+                        style={{padding: "100px"}}>
```
[the remark this answers is not in the document]
Comment (Fedirrr) [reply src/components/AboutCompany.tsx:49]: +

File: src/components/Details.tsx (hunk 7 above)
Comment (Fedirrr) [reply src/components/Details.tsx:18]: +

File: src/components/Header.tsx (hunk 8 above)
Comment (Fedirrr) [reply src/components/Header.tsx:51]: +

File: src/components/Home.tsx (hunk 9 above)
Comment (Fedirrr) [reply src/components/Home.tsx:11]: +

File: src/components/Modal.tsx (hunk 10 above)
Comment (Fedirrr) [reply src/components/Modal.tsx:29]: +

File: src/components/SliderDetails.tsx [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+
+                            <sliderDetailsStyles.MovieWrapDescriptionInfo>
+
+                                <sliderDetailsStyles.MovieTitleDescription>
+                                    Description
+                                </sliderDetailsStyles.MovieTitleDescription>
+
+
+                                <sliderDetailsStyles.MovieDescriptionInfo>
+                                    <dl>
```
[the remark this answers is not in the document]
Comment (Fedirrr) [reply src/components/SliderDetails.tsx:59]: its object

File: src/store/slices/watchListSlice.ts (hunk 11 above)
Comment (Fedirrr) [reply src/store/slices/watchListSlice.ts:7]: +

**Review by VladDrawer — APPROVED** [review]
